.pre-commit-config.yaml

```diff
@@ -1,52 +1,60 @@
----
+# See https://pre-commit.com for more information
+# See https://pre-commit.com/hooks.html for more hooks
 repos:
-  - repo: https://github.com/python/black
-    rev: 23.3.0
+  - repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v4.4.0
     hooks:
-      - id: black
+      - id: check-symlinks
+      - id: destroyed-symlinks
+      - id: trailing-whitespace
+      - id: end-of-file-fixer
+      - id: check-yaml
+      - id: check-toml
+      - id: check-ast
+      - id: check-added-large-files
+      - id: check-merge-conflict
+      - id: check-executables-have-shebangs
+      - id: check-shebang-scripts-are-executable
+      - id: detect-private-key
+      - id: debug-statements
   - repo: https://github.com/codespell-project/codespell
     rev: v2.2.4
     hooks:
       - id: codespell
-        args:
-          - --skip=*.css,*.js,*.map,*.scss,*.svg,*.ipynb
-          - --ignore-words-list=magent
+        exclude: metaworld/envs/assets_updated/sawyer_xyz/dm_control_pick_place.ipynb
+#        args:
+#          - --ignore-words-list==
   - repo: https://github.com/PyCQA/flake8
     rev: 6.0.0
     hooks:
       - id: flake8
         args:
           - '--per-file-ignores=*/__init__.py:F401'
-          - --extend-ignore=E203,W605,F841,E731,E741,F401 # TODO: fix some of these
-          - --max-complexity=205
-          - --max-line-length=300
+          - --ignore=E203,W503,E741
+          - --max-complexity=30
+          - --max-line-length=456
           - --show-source
           - --statistics
-  - repo: https://github.com/PyCQA/isort
-    rev: 5.12.0
-    hooks:
-      - id: isort
-        args: ["--profile", "black"]
   - repo: https://github.com/asottile/pyupgrade
-    rev: v3.3.1
+    rev: v3.3.2
     hooks:
       - id: pyupgrade
-        args: ["--py37-plus"]
-  - repo: https://github.com/pre-commit/pre-commit-hooks
-    rev: v4.4.0
+        args: ["--py38-plus"]
+  - repo: https://github.com/PyCQA/isort
+    rev: 5.12.0
     hooks:
-      - id: mixed-line-ending
```

Comment:
Is this not used in gymnasium? Seems useful, have had issues with line endings before. Also I feel like we should use the same pre commit config across farama projects similar to farama notifications and the furo documentation stuff. Might go and add these things to pettingzoo. For consistency’s sake.

Reply:
I believe that this is done in the `end-of-file-fixer` within `pre-commit-hooks`